### PR TITLE
connect: regression testing bugfix

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message: n/a
Additional Description: n/a
Risk Level: n/a (test only)
Testing: yep
tests  https://github.com/envoyproxy/envoy/issues/34096

### CODE REVIEW

File: source/extensions/transport_sockets/http_11_proxy/connect.h
Comment (ravenblackx) [inline source/extensions/transport_sockets/http_11_proxy/connect.h:85]: Why not put this operation into the `status200()` function rather than precalculating and storing? (If there is a reason, a comment explaining that reason would be good.)

File: test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc
Comment (ravenblackx) [inline test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc:130]: Super-nitty since it's just a test, but `absl::string_view` is good for this.

File: test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc
Comment (ravenblackx) [inline test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc:456]: Should the test *without* content length also still exist?

File: source/extensions/transport_sockets/http_11_proxy/connect.h
Comment (ravenblackx) [reply source/extensions/transport_sockets/http_11_proxy/connect.h:85]: (I see the reason in #34096 now, so explanatory comment is the request!)

**Review by ravenblackx — DISMISSED** [review]

**Comment by RenjieTang** [comment]
I was thinking maybe the fix should be in Envoy's BalsaParser implementation?

It kinda scares me that after the parser finishes executing an input, the headers_ object is cleared. Maybe the headers should just stay?

**Comment by alyssawilk** [comment]
yeah said on ucn chat, I'm hoping we can land a real fix in google3 ASAP at which point I'll put this on hold, and use it to regression test after import lands.
If we can't get a balsa fix in by wednesday we can patch this workaround in as a temporary thing.

**Comment by alyssawilk** [comment]
balsa fix landed, so this is now a test-only change

File: test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc
Comment (ravenblackx) [reply test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc:456]: Didn't get a response on this comment. Seems like both paths should be getting tested.

**Review by ravenblackx — APPROVED** [review]

File: test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc
Comment (alyssawilk) [reply test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc:456]: yeah sorry we do that by default - all th eother tests test this with  stripConnectUpgradeAndRespond()

**Pull request merged**
